model-optimizer/extensions/middle/ONNXResize11ToInterpolateV4.py

```diff
@@ -0,0 +1,166 @@
+"""
+ Copyright (C) 2020 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import logging as log
+import numpy as np
+
+from extensions.ops.Cast import Cast
+from extensions.ops.elementwise import Add, Div, Mul
+from extensions.ops.interpolate import Interpolate
+from mo.front.common.layout import get_depth_dim, get_height_dim, get_width_dim
+from mo.front.common.partial_infer.utils import int64_array, float_array
+from mo.front.tf.graph_utils import create_op_with_const_inputs
+from mo.middle.replacement import MiddleReplacementPattern
+from mo.graph.graph import Graph, Node, rename_nodes
+from mo.ops.const import Const
+from mo.ops.shape import Shape
+from mo.ops.strided_slice import StridedSlice
+
+
+def convert_mode(onnx_mode: str) -> str:
+    return {'nearest': 'nearest', 'linear': 'linear_onnx', 'cubic': 'cubic'}[onnx_mode]
+
+
+def replace_resize(graph: Graph, resize: Node):
+    log.debug("Converting of ONNX Resize-11 to Interpolate-4 is triggered.")
+
+    input_shape = resize.in_port(0).data.get_shape()
+    input_rank = len(input_shape)
+    if input_rank not in {4, 5}:
+        log.warning('The input shape is not 4D or 5D for op with name {}'.format(resize.name))
+        return
+
+    num_of_inputs = len(resize.in_ports())
+    resize_name = resize.name
+    assert num_of_inputs in {3, 4}, \
+        "Number of inputs of ONNXResize (with name {}) should be equal to 3 or 4".format(resize_name)
+
+    assert resize.soft_get('coordinate_transformation_mode') != 'tf_crop_and_resize', \
```

Comment:
We actually limitedly support TF CropAndResize operation so we should make another transformation here instead of throwing an exception. Refer to `model-optimizer/extensions/front/tf/CropAndResizeReplacement.py`

Reply:
Done.

Reply:
I don't understand why this comment has been resolved. I expect to see replacing of Resize-11 with `'coordinate_transformation_mode') == 'tf_crop_and_resize'` to ROIPooling operation. Please, open a ticket to make this change in the future release.

Reply:
Because it can be `coordinate_transformation_mode = tf_crop_and_resize`, and various combination of attributes `mode`, `nearest_mode`, `cubic_coeff_a`. For example, it can be (according to https://github.com/onnx/onnx/blob/rel-1.7.0/docs/Operators.md#Resize) `coordinate_transformation_mode = tf_crop_and_resize`, `mode = nearest`, `nearest_mode = round_prefer_ceil`, or `coordinate_transformation_mode = tf_crop_and_resize`, `mode = cubic`, , `cubic_coeff_a = -0.47`, and so on.  Hence, in the case `coordinate_transformation_mode = tf_crop_and_resize` we should replace Resize-11 with the pair `ROIPooling + Interpolate-4`.